crates/sidecar/src/main.rs

```diff
@@ -28,14 +28,38 @@ async fn run(
 ) -> anyhow::Result<()> {
     let connection_info = ConnectionInfo::from_path(connection_file_path).await?;
 
-    let mut iopub_connection = connection_info
-        .create_client_iopub_connection("", "sidecar-session")
-        .await?; // todo: generate session ID
+    let mut iopub = connection_info
+        .create_client_iopub_connection("", &format!("sidecar-{}", uuid::Uuid::new_v4()))
+        .await?;
+
+    let mut shell = connection_info
+        .create_client_shell_connection(&iopub.session_id)
+        .await?;
+
+    let (tx, mut rx) = futures::channel::mpsc::channel::<JupyterMessage>(100);
 
-    let _webview = WebViewBuilder::new(&window)
+    smol::spawn(async move {
+        while let Some(message) = rx.next().await {
+            if let Err(e) = shell.send(message).await {
+                eprintln!("Failed to send message: {}", e);
+                break;
+            }
+        }
+    })
+    .detach();
```

Comment:
We will probably need to process the shell receives as well, though that will run into issues since the socket can't be split at this moment to get around borrow checker constraints.